### ISSUE TITLE
Large Mol. SC Compartment Mobi

### ISSUE THREAD

**Issue by prvmalik** [Open-Systems-Pharmacology/Forum#147]
Hi all.  I'm creating a simple subcutaneous absorption compartment for the large molecule model in Mobi.  Here's what I've done:

Add spatial structure:  Subcutaneous (organ) - Can contain molecules, V = 1
Add spacial structure:  SubcutaneousClearance (compartment) - Can contain molecules, V = 1
Create neighbourhood:  Subcutaneous_VenousBlood_pls
Create neighbourhood:  Subcutaneous_SubcutaneousClearance
Add molecule parameter (global):  Ka, inversed time
Add molecule parameter (global):  Ke, inversed time
Add passive transport:  SubcutaneousAbsorption, Source Subcutaneous, Target VenousBlood (Venous Blood plasma was not available)
Add formula: SubcutaneousAbsorption, = Ka x Concentration (referenced to Subcutaneous)
Add passive transport:  Subcutaneous Clearance, Source Subcutaneous, Target SubcutaneousClearance
Add formula: SubcutaneousClearnace, = Ke x Concentration (referenced to Subcutaneous)
Turned all molecule start values for Subcutaneous and Subcutaneous clearance spatial structures to "Not Present" except for the drug molecule (i.e. no FcRn, target, endoIgG, etc.)

When I ran the simulation, I got an error that there were negative values.

Adapting the molecule start values for the Subcutaneous compartment to allow negative values allows the simulation to run, however it is an ever-increasing function.

Let me know if there are any major errors or omissions that would explain this.

**Comment by Yuri05** [comment]
Is it possible to attach your project?
Also what do you mean by "_Venous Blood plasma was not available_"?

**Comment by prvmalik** [comment]
Thanks for your response Yuri.  My project file is attached.

When I say that Venous Blood Plasma was not available, I mean that when I was assigning the passive transport target compartment by creating a "new match tag condition" in the target box, Venous_Blood_pls was not an option that I could select, so I selected Venous_Blood instead.

[40 mg SC.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/1807687/40.mg.SC.zip)

**Comment by Yuri05** [comment]
>When I say that Venous Blood Plasma was not available, I mean that when I was assigning the passive transport target compartment by creating a "new match tag condition" in the target box

For that, you must set **2** match conditions:
1. _Venous blood_
2. _Plasma_
<img width="400" alt="tag" src="https://user-images.githubusercontent.com/25061876/37414512-3566d5ae-27a1-11e8-8812-800f8587c41d.png">

Regarding negative values: the error is in the definition of the application. 
Your project (as every project exported from PK-Sim) is amount based. 
Means: internally MoBi works with molecule **amounts**; concentrations are just _derived_ via
`C = Amount/Volume`

So you must set the initial **amount** of _Subcutaneous|Adalimumab_ and not its concentration
(correct Changed Entity Path would be `Organism|Subcutaneous|Adalimumab`)

What happens in your case: by setting the initial **concentration** the relationship `C = Amount/Volume` is destroyed and the concentration is set to a constant value which **does not change during the simulation**.
For the transport `Subcutaneous|Adalimumab => Venous Blood|Plasma` it means: the transport rate becomes a constant positive value. Which leads to negative values of `Subcutaneous|Adalimumab
`
<img width="464" alt="aufnahme26" src="https://user-images.githubusercontent.com/25061876/37414686-9b56980e-27a1-11e8-8452-a497dae1ba2d.png">

**Comment by msevestre** [comment]
@Yuri05 Great find. Would it be possible to actually check that dimensions are consistent? One can see clearly in your screenshot that the Formula is DrugMass (Amount) and is assigned to a Concentration. I would expect to at least see a warning when dimensions are not consistent. Thoughts?

**Comment by Yuri05** [comment]
Currently all warnings are shown during simulation creation. But those for event assignments are missing indeed. Would be helpful.

@prvmalik Checking the warnings while creating simulation you can also see that both transports _SubcutaneousAbsorption_ and _SubcutaneousClearance_ are not properly set: both define change of _concentration_ and not of the _amount_.
(numerically it does not matter as long as your Volume is 1)

**Comment by prvmalik** [comment]
Thank you very much for your clear responses and instructions.  The model works great now.

I thought I could get away with using concentrations because the V was 1.  I didn't know that the compound name represented the amount in the selection tree.  I changed the concentrations to amounts in the passive transport formulae and I updated the event initial conditions to set the subcutaneous adalimumab amount, rather than concentration.
